# Linear Regression <br> Regression Diagnostics

Instructions:


Load the dataset named Auto (in the ISLR package) into R.

 

Background Information

The dataset Auto contains the fuel efficiency and other characteristics of 392 cars that were produced between 1970 and 1982. The variables have the following definitions:

 

- mpg: The fuel efficiency of the car, measured in miles per gallon,

- cylinders: The number of cylinders in the engine of the car,

- displacement: The displacement of the car’s engine, measured in cubic inches,

- horsepower: The power of the car’s engine, measured in horsepower,

- weight: The weight of the car, measured in pounds,

- acceleration: The quarter mile drag time, measured in seconds,

- year: The last two digits of the car’s model year,

- origin: The origin region of the car. 1 stands for U.S., 2 stands for Europe, 3 stands for Japan,

- name: The make and model of the car.

In [1]:
library(ISLR)
library(car)
head(Auto)

Warning message:
"package 'ISLR' was built under R version 3.4.4"Warning message:
"package 'car' was built under R version 3.4.4"Loading required package: carData
Warning message:
"package 'carData' was built under R version 3.4.4"

mpg cylinders displacement horsepower weight acceleration year origin
1 18  8         307          130        3504   12.0         70   1     
2 15  8         350          165        3693   11.5         70   1     
3 18  8         318          150        3436   11.0         70   1     
4 16  8         304          150        3433   12.0         70   1     
5 17  8         302          140        3449   10.5         70   1     
6 15  8         429          198        4341   10.0         70   1     
  name                     
1 chevrolet chevelle malibu
2 buick skylark 320        
3 plymouth satellite       
4 amc rebel sst            
5 ford torino              
6 ford galaxie 500

** Use the summary function to  **
** find the average value of the following variables: mpg, cylinders, horsepower, weight, acceleration, year **

In [2]:
summary(subset(Auto, select=c('mpg', 'cylinders', 'horsepower', 'weight', 'acceleration', 'year')))[4,]


mpg         cylinders        horsepower            weight 
"Mean   :23.45  " "Mean   :5.472  " "Mean   :104.5  "  "Mean   :2978  " 
     acceleration              year 
"Mean   :15.54  " "Mean   :75.98  "

** Use the correlation function to find the two numeric variables (mpg, cylinders, horsepower, weight, acceleration, year) **
** with the maximum negative correlation.**

In [3]:
df<-cor(subset(Auto, select=c('mpg', 'cylinders', 'horsepower', 'weight', 'acceleration', 'year')))
min(df)
df

[1] -0.8322442

mpg        cylinders  horsepower weight     acceleration
mpg           1.0000000 -0.7776175 -0.7784268 -0.8322442  0.4233285  
cylinders    -0.7776175  1.0000000  0.8429834  0.8975273 -0.5046834  
horsepower   -0.7784268  0.8429834  1.0000000  0.8645377 -0.6891955  
weight       -0.8322442  0.8975273  0.8645377  1.0000000 -0.4168392  
acceleration  0.4233285 -0.5046834 -0.6891955 -0.4168392  1.0000000  
year          0.5805410 -0.3456474 -0.4163615 -0.3091199  0.2903161  
             year      
mpg           0.5805410
cylinders    -0.3456474
horsepower   -0.4163615
weight       -0.3091199
acceleration  0.2903161
year          1.0000000

** Please estimate a linear regression model (using the lm function) with mpg as the dependent variable and horsepower as the independent variables. **


In [4]:
lm.1<-lm(formula= mpg~horsepower,data=Auto)
summary(lm.1)


Call:
lm(formula = mpg ~ horsepower, data = Auto)

Residuals:
     Min       1Q   Median       3Q      Max 
-13.5710  -3.2592  -0.3435   2.7630  16.9240 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)    
(Intercept) 39.935861   0.717499   55.66   <2e-16 ***
horsepower  -0.157845   0.006446  -24.49   <2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 4.906 on 390 degrees of freedom
Multiple R-squared:  0.6059,	Adjusted R-squared:  0.6049 
F-statistic: 599.7 on 1 and 390 DF,  p-value: < 2.2e-16


- ** What is this model’s R-squared value?  0.6059 **


- ** What is this model’s adjusted R-squared value?  0.6049 **

- ** What is the estimated horsepower coefficient? -0.157845 **

- ** What is the F-statistic of the overall model? 599.7 **


- ** what are the values of Total Sum of Squares (SST), Sum of Squared Errors (SSE), and Sum of Squared Regression (SSR)? **


In [5]:
anova(lm.1)

Df  Sum Sq    Mean Sq     F value  Pr(>F)      
horsepower   1 14433.078 14433.07760 599.7177 7.031989e-81
Residuals  390  9385.916    24.06645       NA           NA

** From above **

** Total Sum of Squares (SSR)  -> 14433.1 **

** Sum of Squared Errors (SSE) -->9385.9 **




In [6]:
SSR<-14433.078
SSE<-9385.916
SST<-SSR+SSE
SST

[1] 23818.99

- ** What is the predicted value of mpg for a car that has a horsepower of 101? **

In [7]:
predict.lm(lm.1,data.frame(horsepower=101), interval = "predict")

fit      lwr      upr     
1 23.99354 14.33611 33.65098

- ** What is the 95% prediction interval of mpg for a car that has a horsepower of 101? **

** 14.33611	33.65098 is 95% range **



Please estimate a linear regression model (using the lm function) with mpg as the dependent variable and 
horsepower, cylinders, weight, acceleration, and year as the independent variables

In [8]:
lm.2<-lm(formula= mpg~horsepower+cylinders+weight+acceleration+year,data=Auto)
summary(lm.2)


Call:
lm(formula = mpg ~ horsepower + cylinders + weight + acceleration + 
    year, data = Auto)

Residuals:
    Min      1Q  Median      3Q     Max 
-8.7973 -2.3663 -0.0524  1.9950 14.3169 

Coefficients:
               Estimate Std. Error t value Pr(>|t|)    
(Intercept)  -1.510e+01  4.734e+00  -3.189  0.00154 ** 
horsepower    3.038e-03  1.344e-02   0.226  0.82129    
cylinders    -9.763e-02  2.466e-01  -0.396  0.69233    
weight       -6.486e-03  6.015e-04 -10.784  < 2e-16 ***
acceleration  7.346e-02  1.014e-01   0.724  0.46928    
year          7.493e-01  5.248e-02  14.278  < 2e-16 ***
---
Signif. codes:  0 '***' 0.001 '**' 0.01 '*' 0.05 '.' 0.1 ' ' 1

Residual standard error: 3.436 on 386 degrees of freedom
Multiple R-squared:  0.8087,	Adjusted R-squared:  0.8062 
F-statistic: 326.4 on 5 and 386 DF,  p-value: < 2.2e-16


- ** What is this model’s R-squared value?  0.8087 **

- ** What is this model’s adjusted R-squared value? 0.8062 **

- ** List all the predictors that appear to have a relationship to mpg that are statistically significant at 5%? weight,year **       



Please use the predict.lm function in R to make prediction for the mpg for a car that has a horsepower of 104.5, 
weight of 3000, 4 cylinders, a model year of 1980, and an acceleration time of 16 seconds..

In [9]:
predict.lm(lm.2,data.frame(horsepower=104.5,cylinders=4,weight=3000,acceleration=16,year=80), interval = "predict")

fit     lwr      upr     
1 26.4906 19.67986 33.30134

- ** We used mpg as dependent variable and horsepower, cylinders, weight, acceleration, and year as explanatory variables to run a regression.<br> Which of the independent variables has the highest Variance Inflation Factor (VIF)? **


In [10]:
vif(lm.2)

horsepower    cylinders       weight acceleration         year 
    8.867796     5.859121     8.646446     2.593284     1.238072